### ISSUE TITLE
Recycling Bin Management

### ISSUE THREAD

**Issue by markus101** [Sonarr/Sonarr#1984]
A page in the UI for managing the recycling bin, to see which files are there, the ability to empty it or restore a file from it back to the library (manual import-like?).

**Comment by RealTabman** [comment]
Speaking of recycle bins, the option to have a separate recycling bin directory for different drives would also be nice.

**Comment by leram84** [comment]
i would also add the ability to handle cleanup automatically... So, set a size limit on the bin (1TB for example) and then keep only the most recent 1TB of files and delete everything else.

**Comment by Thirrian** [comment]
I like the idea of a size limit on the recycle bin, but it's probably a lot of code to write (I don't know, I'm not a dev).

Would you consider adding the "delete everything older than 7 days in the recycle bin" interval as a setting in the GUI?
(advanced option, don't allow users to set it to 0 or higher than a reasonable maximum, default = 7)
Use case: I'd like to set the interval a bit shorter. My disk still fills up with lower quality files that were upgraded but I'm reluctant to turn off the recycle bin altogether since I occasionally have to "save" a file in it.

Somewhat related: the tooltip on the recycle bin option says "files will go here [...] instead of being permanently deleted". But after 7 days, they're deleted anyway (!)
Haven't seen anyone complain about that, but e.g. in Windows you can leave files in the recycle bin forever if you really want. I think maybe a little warning for people thinking Sonarr's recycle bin works the same way would be nice.

**Comment by Kyrluckechuck** [comment]
+1 to this, totally forgot I had a recycle folder setup for my Sonarr installation recently and had to delete 2.4TB of data..

**Comment by eriksneff** [comment]
@Kyrluckechuck me too! That's why I'm here. I use radarr and sonarr on macos, and had configured them to put removed items into the mac's /Volumes/drivename/.Trashes folder, and didn't think twice. Then I noticed I was running out of space, so I did the mac dock empty recycle bin process I'd normally do, and that didn't help. I then hopped into the terminal and low and behold, the /Volumes/drivename/.Trashes folder was not empty! Yikes. Will be unsetting the recycle bin path until I can find a more usable way to manage deleted items.

**Comment by CollinChaffin** [comment]
I would still like to see something in place to manage it.  TIA!

**Comment by tmchow** [comment]
I'm surprised there's not limit on the diskspace used or some automated job to clean it up old files.  So count me in for adding my vote on this feature request :)

**Comment by runaroundhome** [comment]
v3 apears to be auto deleting older then 1 week files. Though I have no idea if it acutly does it, I just saw it in the change log (I'm hoping there will be a way to customize the 1 week and/or disable it).
https://github.com/Sonarr/Sonarr/commit/dc7b4cebf2b0990e95aed2a96300f1dea1b13390

**Comment by Thirrian** [comment]
Interval is configurable in v3: https://github.com/Sonarr/Sonarr/commit/8196f6b9dbe5b5f3a76b236d5df695e4762e6571

**Comment by runaroundhome** [comment]
Oh wow, I totally missed that change last week, thank you @Thirrian !!

**Comment by robert-corlett** [comment]
Yes a UI would be nice... Or at least if Radarr could stop getting over excited and deleting movies before it has actually finished importing and checking the new movie that would be nice then I wouldnt have to bother with the recycling bin :-)

**Comment by Panzer1119** [comment]
Is there any progress on this?

I think this is a very handy feature.

**Comment by a3gill** [comment]
> Speaking of recycle bins, the option to have a separate recycling bin directory for different drives would also be nice.

This is what I came looking for. When I have root folders on different drives, the value drops when having to move to the other drive just to bin it.

An alternate solution could be to allow relative paths. For example, if you have a library setup at `/mnt/media/libraries/tv` and `/mnt/media2/libraries/tv` then a path relative to the library root could be `../../trash/tv` -- just thinking out loud.

**Comment by uchagani** [comment]
One thing i'd like to see is an option to configure which series uses the recycle bin.  For example, when adding/editing a series, a checkbox that says "use recycle bin".  This way, i only have to use up space for series i really care about keeping long term in case of a bad grab.

**Comment by bbtufty** [comment]
> > Speaking of recycle bins, the option to have a separate recycling bin directory for different drives would also be nice.
> 
> This is what I came looking for. When I have root folders on different drives, the value drops when having to move to the other drive just to bin it.

+1 to this, I've got things working downloading to different volumes via tags now, so the slowest part of the import is the recycle. Having an option to either map relatively would be good, or even more generally a recycle bin for each root folder would allow for more unusual directory structures

**Comment by sunnyd24** [comment]
**@markus101 Is there a plan to add this to a future milestone since it was removed on Jul 31?**

Having a UI page dedicated to manage recycling (for easy review & restore, etc.) when something goes wrong would be very useful.

E.g. I just grabbed a release using interactive search, and it advised a manual import was required as it couldn't match the details to the season, however, I found it automatically imported and assigned the files to the wrong episodes as the release was not correct. 
I can't seem to find a way to roll back to old files anymore.............this UI would help in situations like this!
